query/src/test/java/com/connexta/replication/query/QueryServiceTest.java

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright (c) Connexta
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+
+package com.connexta.replication.query;
+
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.notNullValue;
+import static org.hamcrest.Matchers.nullValue;
+import static org.junit.Assert.*;
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import com.connexta.replication.api.NodeAdapter;
+import com.connexta.replication.api.data.Filter;
+import com.connexta.replication.api.data.FilterIndex;
+import com.connexta.replication.api.data.Metadata;
+import com.connexta.replication.api.data.QueryRequest;
+import com.connexta.replication.api.data.QueryResponse;
+import com.connexta.replication.api.data.Site;
+import com.connexta.replication.api.queue.SiteQueue;
+import java.time.Instant;
+import java.util.Date;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Stream;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public class QueryServiceTest {
```

Comment:
❗️ We should definitely create test cases that verifies the behavior of being stopped by the scheduler via interruptions when we are querying, and when we are creating and queuing tasks to make sure we keep our index up to date.